### ISSUE TITLE
Hangs forever on chained functions

### ISSUE THREAD

**Issue by garamond** [reactjs/react-docgen#136]
Input file `MyComp.js`:
```
export class MyComp extends Component { render(){} }

export function myFun(foo) {
  return foo.bar().baz()
}
```

Running `react-docgen MyComp.js` hangs forever and causes CPU load to spike to 100%.

Without chained functions the issue does not appear: After changing to body of `myFun` to
```
export function myFun(foo) {
  return foo.bar.baz()
}
```
the file is processed without issues.

**Comment by fkling** [comment]
Thanks! This will be resolved with #134 .

**Comment by tchon** [comment]
@fkling It looks like https://github.com/reactjs/react-docgen/pull/134 will not address the hanging on chained methods. See improved PR here: https://github.com/reactjs/react-docgen/pull/139
